### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by DuongTyler**
Rewrite of the original frontend using functional react and chakra-ui

### CODE REVIEW

**Comment by Nephren68** [comment]
👀

**Comment by DuongTyler** [comment]
image

File: backend-next/migrations/0000-initial.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CREATE TABLE IF NOT EXISTS messages (
+  id uuid primary key default gen_random_uuid(),
+  created_at timestamp with time zone NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  username varchar(128) NOT NULL,
+  orig_msg text NOT NULL,
+  tl_msg text,
+  country varchar(2)
```
Comment (zapplecat) [inline backend-next/migrations/0000-initial.sql:7]: I forget, is there a possibility where someone didn't have a country? Did we put antartica for those folks or something?

File: backend-next/migrations/0000-initial.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS games (
+  id uuid primary key default gen_random_uuid(),
+  created_at timestamp with time zone NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  title varchar(256) NOT NULL,
+  game_description text NOT NULL,
+  git_link text,
+  game_link text NOT NULL,
+  thumbnail text
```
Comment (zapplecat) [inline backend-next/migrations/0000-initial.sql:26]: Is this actually an URL in some cases? I forget

File: backend-next/migrations/migrate_data.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    res = requests.get("https://manoaloe.manotomo.com/api/gallery")
+    gallery = json.loads(res.text)
+    return gallery.get("gallery")
+
+
+def get_games() -> List[Game]:
+    res = requests.get("https://manoaloe.manotomo.com/api/games")
+    games = json.loads(res.text)
+    return games.get("games")
+
```
Comment (zapplecat) [inline backend-next/migrations/migrate_data.py:51]: Maybe do `assert res.ok` for all of these just in case.

File: frontend-next/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        path: "/home",
+        element: <Home />,
+      },
+      {
+        path: "/game",
+        element: <Games />,
+      },
+      {
+        path: "/art",
+        element: <Gallery />,
```
Comment (zapplecat) [inline frontend-next/src/App.tsx:98]: So weird that we have `/art` paired with a `Gallery` element. /shrug

File: backend-next/migrations/0000-initial.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
+CREATE TABLE IF NOT EXISTS messages (
+  id uuid primary key default gen_random_uuid(),
+  created_at timestamp with time zone NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  username varchar(128) NOT NULL,
+  orig_msg text NOT NULL,
+  tl_msg text,
+  country varchar(2)
+);
+
+CREATE TABLE IF NOT EXISTS gallery (
```
Comment (zapplecat) [inline backend-next/migrations/0000-initial.sql:10]: What happened to the blurhash? Granted, I guess that makes things more complex, and we might not even really care...

EDIT: Oh, I see you're using skeleton later. That works.

File: frontend-next/src/credits.tsx [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    return (
+        <Box>
+            <Heading size="md">
+                Website Team <br />
+            </Heading>
+<br />
+            <Text>
+                Kyuubei - Web Design <br />
+                Jesse Zong - Frontend Website Designer <br />
+                Zapple - Backend Development & Frontend contributions <br />
```
Comment (zapplecat) [inline frontend-next/src/credits.tsx:13]: Does it not like the `&` here? Do you need to escape it?

**Review by zapplecat — APPROVED** [review]
Minor comments, lgtm

**Comment by zapplecat** [comment]
Oh yeah, can you add some notes about how to actually get things up and running locally?

File: backend-next/migrations/0000-initial.sql (hunk 1 above)
Comment (DuongTyler) [reply backend-next/migrations/0000-initial.sql:7]: We just left them blank lol

File: backend-next/migrations/0000-initial.sql (hunk 2 above)
Comment (DuongTyler) [reply backend-next/migrations/0000-initial.sql:26]: yep, it's a url :D

File: frontend-next/src/credits.tsx (hunk 6 above)
Comment (DuongTyler) [reply frontend-next/src/credits.tsx:13]: /shrug

File: backend-next/migrations/0000-initial.sql (hunk 5 above)
Comment (DuongTyler) [reply backend-next/migrations/0000-initial.sql:10]: blurhash didn't get ported over to the new db since we have skeleton yea

**Pull request merged**